services/forms/user_form.go

```diff
@@ -262,6 +262,92 @@ func (f *UpdateLanguageForm) Validate(req *http.Request, errs binding.Errors) bi
 	return middleware.Validate(errs, ctx.Data, f, ctx.Locale)
 }
 
+// UpdateCommentTypeForm form for updating profile
+type UpdateCommentTypeForm struct {
+	Reference    bool
+	Labels       bool
+	Milestone    bool
+	Assignee     bool
+	Title        bool
+	Branch       bool // delete branch
+	Time         bool
+	Deadline     bool
+	Dependencies bool
+	Lock         bool
+	Target       bool // target branch
+	Requests     bool // review requests
+	Push         bool // push to PR
+	Project      bool
+	Ref          bool // issue ref
+}
+
+// Validate validates the fields
+func (f *UpdateCommentTypeForm) Validate(req *http.Request, errs binding.Errors) binding.Errors {
+	ctx := context.GetContext(req)
+	return middleware.Validate(errs, ctx.Data, f, ctx.Locale)
+}
+
+// IsHidden checks if the given commentType is visible
+func (f *UpdateCommentTypeForm) IsHidden(commentType int) bool {
+	switch commentType {
+	case 3:
```

Comment:
Here and below, should we use the consts like `CommentTypeIssueRef`?

Magic numbers are really hard to maintain.

Reply:
I know, but I had to use them because of the settings check which can't access these comment types from the `models` pkg

Reply:
https://docs.gitea.io/en-us/guidelines-backend/

Accessing models from services should work.

Reply:
The problem is, I need it in a template...